tests/bin/demo-addition.test.ts

```diff
@@ -0,0 +1,55 @@
+import main from '@/bin/demo-addition';
+
+const maybeMockRestore = (a: any): void =>
+  a.mockRestore && typeof a.mockRestore === 'function'
+    ? a.mockRestore()
+    : undefined;
+
+type FunctionPropertyNames<T> = {
+  [K in keyof T]: T[K] extends (...args: any[]) => any ? K : never;
+}[keyof T];
+type ArgsType<T> = T extends (...args: infer A) => any ? A : never;
+
+function spyOnImplementing<
+  T extends typeof console,
+  M extends FunctionPropertyNames<T>,
+  F extends T[M],
+  I extends (...args: any[]) => ReturnType<F>
+>(
+  target: T,
+  property: M,
+  impl: I,
+): jest.SpyInstance<ReturnType<F>, ArgsType<F>> {
+  maybeMockRestore(target[property]);
+  return jest.spyOn(target, property as any).mockImplementation(impl);
+}
+
+const mockConsoleLog = () => spyOnImplementing(console, 'log', (() => ({})));
+
+describe('main', () => {
+  test('main takes synthetic parameters', () => {
+    // jest can also "spy on" the console object
+    // and then you can test on stdout
+    expect(main(['1'])).toEqual(0);
+  });
+  test('adds 0 + 0', () => {
+    const mock00 = mockConsoleLog();
+    main(['thing', 'thing', '0', '0']);
+    expect(mock00).toHaveBeenCalledWith('0 + 0 = 0');
+  });
+  test('adds 0 + 1', () => {
+    const mock01 = mockConsoleLog();
+    main(['thing', 'thing', '0', '1']);
+    expect(mock01).toHaveBeenCalledWith('0 + 1 = 1');
```

Comment:
Try this instead:

```
test('...', () => {
  const mockLog = mockConsoleLog();
  main([...]);
  expect(mockLog).toHaveBeenCalledWith('Browser log');
  mockLog.mockRestore();
});
```

The above is for when console.log is used.

Then change your script to using `process.stdout.write("...\n");` instead.

Then use the `mockProcessStdout();` instead.